src/java/voldemort/client/protocol/vold/VoldemortNativeClientRequestFormat.java

```diff
@@ -116,7 +116,27 @@ public void writeGetRequest(DataOutputStream outputStream,
     }
 
     public boolean isCompleteGetResponse(ByteBuffer buffer) {
-        return isCompleteResponse(buffer, VoldemortOpCode.GET_OP_CODE);
+        try {
+            DataInputStream inputStream = new DataInputStream(new ByteBufferBackedInputStream(buffer));
+            checkException(inputStream);
+            if(!skipeResults(inputStream, buffer)) {
+                return false;
+            }
+        } catch(VoldemortException e) {
+        } catch(IOException e) {
+            return false;
+        }
+        return !buffer.hasRemaining();
+    }
+
+    private boolean skipeResults(DataInputStream inputStream, ByteBuffer buffer) throws IOException {
```

Comment:
Vinoth, I will add the javadocs comments. Sorry for not commenting it. All of the work here is done in my personal time and I made a case why we need these fixes which was fortunately approved. So I could not spend enough time on it as I would like it. But that should not be an excuse and I will fix it in a separate commit.